src/App.tsx

```diff
@@ -1,16 +1,46 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import cn from 'classnames';
+import debounce from 'lodash.debounce'
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [isDroppedDown, setIsDroppedDown] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('')
+  const [appliedQuery, setAppliedQuery] = useState('')
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), [])
+
+  const personClickHandler = (person: Person) => {
+    setIsDroppedDown(false)
+    setSelectedPerson(person)
+    setQuery(person.name)
+  }
+
+  const inputOnChangeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value)
+    applyQuery(event.target.value)
+    setSelectedPerson(null)
+  }
+
+  const filteredPeople = peopleFromServer.filter(person => {
+    return person.name.toLowerCase().includes(appliedQuery.toLowerCase())
+  })
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
-        <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
-        </h1>
+        {selectedPerson ? (
+          <h1 className="title" data-cy="title">
+          {`${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`}
+          </h1>
+        ) : (
+          <h1 className="title" data-cy="title">
+            No selected person
+          </h1>
+        )}
```

Comment:
Why do you re-render the whole `h1` ?

```suggestion
         <h1 className="title" data-cy="title">
                {selectedPerson ? `${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})` : 'No selected person'}
          </h1>
        )}
```